In [16]:
from math import sqrt
import pandas as pd
t_table = pd.read_csv('../data/t_table.csv', header=0, index_col=0)

# 蝶の体長問題
x = [76, 85, 82, 83, 76, 78]
x_num = len(x)

# 標本平均
x_ave = 1 / x_num * sum(x)
print('標本平均: {}'.format(x_ave))

# 標本分散
s_2 = 1 / x_num * sum([(e - x_ave)**2 for e in x])
print('標本分散: {}'.format(s_2))

# 標本標準偏差
s = sqrt(s_2)
print('標本標準偏差: {}'.format(s))

# t分布の自由度
dof = x_num - 1

# その時の限界値
limit_value = t_table.at[dof, 'limit_value']
print('限界値(自由度{}): {}'.format(dof, limit_value))

標本平均: 80.0
標本分散: 12.333333333333332
標本標準偏差: 3.511884584284246
限界値(自由度5): 2.571


これらから，

$$
\begin{array} \\
-2.571 <= \displaystyle \frac{(80 - \mu)\sqrt{5}}{3.51} <= 2.571 \\
-9.024 <= (80 - \mu) \sqrt{5} <= 9.024 \\
-4.035 <= 80 - \mu <= 4.035 \\
75.96 <= \mu <= 84.04
\end{array}
$$

- 母分散：W統計量を使ってchi2分布から推定
- 母平均：T統計量を使ってt分布から推定